storage_controller/src/service.rs

```diff
@@ -875,6 +875,22 @@ impl Service {
             });
         }
 
+        // Fetch the list of completed imports and attempt to finalize them in the background.
+        // This handles the case where the previous storage controller instance shut down
+        // whilst finalizing imports.
+        let complete_imports = self.persistence.list_complete_timeline_imports().await;
+        match complete_imports {
+            Ok(ok) => {
+                tokio::task::spawn({
+                    let finalize_imports_self = self.clone();
+                    async move { finalize_imports_self.finalize_timeline_imports(ok).await }
+                });
+            }
+            Err(err) => {
+                tracing::error!("Could not retrieve completed imports from database: {err}");
```

Comment:
observability: Maybe worthy of `critical!` ? This will lead to stuck projects alert at cplane eventually, and having a `critical!` alert in storcon will shorten investigation time.

Reply:
robustness: Also, I was going to ask for a retry loop, but, then realized that we're not keeping track of these spawned `finalize_timeline_imports` task in any way. So, when retrying, we woulnt' know for which tasks we have already launched finalize tasks and for which ones we didn't.